src/Engine/Messages/MessageModel.cs

```diff
@@ -52,6 +52,12 @@ public class MessageModel : ISerializable
         }
 
         [DataMember]
+        public string ID {
```

Comment:
Have you tested that against an older server that has no ID yet? I think the serializer part is also missing since it uses a custom one.